services/datamanager/datacapture/data_capture_file.go

```diff
@@ -297,15 +296,22 @@ func GetAllReadings(filePath string) ([]*v1.SensorData, error) {
 		return nil, err
 	}
 
+	return SensorDataFromFile(dcFile)
+}
+
+// SensorDataFromFile returns all readings in f.
+func SensorDataFromFile(f *File) ([]*v1.SensorData, error) {
+	f.Reset()
+	var ret []*v1.SensorData
 	for {
-		next, err := dcFile.ReadNext()
-		if errors.Is(err, io.EOF) {
-			break
-		}
+		next, err := f.ReadNext()
 		if err != nil {
+			if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
```

Comment:
Thanks for consolidating this with `getAllSensorData`. I imagine it might be tricky to test without introducing sigint channel readings, but would it be possible to include a test for this case?

Reply:
Added a test in data_capture_file_test for this case